### PR TITLE
global-javascript: Adds OneTrust utilities

### PR DESCRIPTION
**Pull request by tulipdexter**
Adds `checkConsent` and `isConsentBannerClosed` utilities.

### CODE REVIEW

File: toolkits/global/packages/global-javascript/README.md [hunk 1]
```diff
@@ -20,10 +20,11 @@ You can import as many of the named exports from the helpers as you require for
 - [getCookie](#getcookie)
 - [getCookie](#debounce)
 - [getCookie](#throttle)
```
Comment (alexkilgour) [inline toolkits/global/packages/global-javascript/README.md:22]: Not part of this PR, but the name are wrong on these!
```
[getCookie](#debounce)
[getCookie](#throttle)
```

Should be:

```
[debounce](#debounce)
[throttle](#throttle)
```

Comment (tulipdexter) [reply toolkits/global/packages/global-javascript/README.md]: 😬

File: toolkits/global/packages/global-javascript/src/helpers/util/onetrust.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	};
+
+	if (!validCategories[category]) {
+		throw new Error('Invalid category: ' + category);
+	}
+
+	const consent = getCookie('OptanonConsent');
+	const consentGroups = consent.split('groups=').pop().split('&')[0];
+	const colonEncoding = '%3A';
+
```
Comment (alexkilgour) [inline toolkits/global/packages/global-javascript/src/helpers/util/onetrust.js:25]: Is it likely that any of this could change? either by us or onetrust. i.e. would it be valuable to make some of this configurable with an `init()` function, such as categories, cookie name? It might be that we hope not, and just cross that bridge when/if it happens, but it's easier for an app to update a config than release a new version of this package every time something changes.

Comment (tulipdexter) [reply toolkits/global/packages/global-javascript/src/helpers/util/onetrust.js:25]: Hmm good question. I think that it's unlikely that the cookie name would change, but it's possible that the categories could change (based on our side). I could make the categories I've set here the default option and allow different categories to be set via an init.

Comment (alexkilgour) [reply toolkits/global/packages/global-javascript/src/helpers/util/onetrust.js:25]: If that's easy then it would safeguard against them changing

Comment (alexkilgour) [reply toolkits/global/packages/global-javascript/src/helpers/util/onetrust.js:25]: After discussion offline, we think it is best to leave it as it is. We want the module to reflect the current state of our OneTrust configuration, and making it configurable would mean those categories _could_ go stale. Every team should be using the same set of categories; if they change for one they change for all. So a new version will be released if the categories change, and teams will update.
